course/src/isometry.cc

```diff
@@ -0,0 +1,102 @@
+#include <cmath>
+#include <cstdint>
+#include "isometry.h"
+
+namespace ekumen {
+namespace math {
+
+Vector3::Vector3(const double& x, const double& y, const double& z) : x_(x), y_(y), z_(z) {}
+
+Vector3::~Vector3() {}
+
+// Class constants
+const Vector3 Vector3::kUnitX = Vector3(1., 0., 0.);
+const Vector3 Vector3::kUnitY = Vector3(0., 1., 0.);
+const Vector3 Vector3::kUnitZ = Vector3(0., 0., 1.);
+const Vector3 Vector3::kZero = Vector3(0., 0., 0.);
+
+Vector3 Vector3::operator+(const Vector3& vec) const {
+    return Vector3(this->x() + vec.x(), this->y() + vec.y(), this->z() + vec.z());
+}
+Vector3 Vector3::operator-(const Vector3& vec) const {
+    return Vector3(this->x() - vec.x(), this->y() - vec.y(), this->z() - vec.z());
+}
+Vector3 Vector3::operator*(const double& value) const {
+    return Vector3(this->x() * value, this->y() * value, this->z() * value);
+}
+Vector3 Vector3::operator*(const Vector3& vec) const {
+    return Vector3(this->x() * vec.x(), this->y() * vec.y(), this->z() * vec.z());
+}
+Vector3 Vector3::operator/(const Vector3& vec) const {
+    return Vector3(this->x() / vec.x(), this->y() / vec.y(), this->z() / vec.z());
+}
+
+const double& Vector3::operator[](const int &index) const {
+    switch (index) {
+        case 0:
+            return this->x();
+        case 1:
+            return this->y();
+        case 2:
+            return this->z();
+        default:
+            throw std::overflow_error("index out of range, Vector3 only have three elements");
+    }
+}
+
+double& Vector3::operator[](const int &index) {
+    switch (index) {
+        case 0:
+            return this->x_;
+        case 1:
+            return this->y_;
+        case 2:
+            return this->z_;
+        default:
+            throw std::overflow_error("index out of range, Vector3 only have three elements");
+    }
+}
+
+bool Vector3::operator==(const std::initializer_list<double>& vector) const {
+    if (vector.size() != 3) return false;
```

Comment:
Don't inline code like this, always use `{ }`. Read [C Traps and Pittfalls](https://www.cs.tufts.edu/comp/40/docs/CTrapsAndPitfalls.pdf) Sec. 2.6 (and the rest of it would be a good idea too).